src/NeinLinq/DynamicExpression.cs

```diff
@@ -119,6 +119,11 @@ public static class DynamicExpression
 
             var expression = (Expression)target;
 
+            if (underlyingType.IsEnum)
+            {
+                return (string? value , IFormatProvider? formatProvider) => value == null ? null : Enum.Parse(underlyingType, value);
```

Comment:
`value is null` (type could override the equals operator, I wouldn't trust those...)